srk/src/transitionSystem.ml

```diff
@@ -662,18 +684,158 @@ module Make
+            if ((PS.cardinal callees) == 0) && ((PPS.cardinal callers) > 0) then PS.add (x, y) acc else acc
+          | (_, _) -> failwith ""
+          ) procedures PS.empty in
+      let copy_subgraph tg src = 
+        let vertices = dfs (-1, -1) tg src (fun _ _ _ -> ()) (ISet.add src ISet.empty) in 
+        let to_map = Hashtbl.create 998 in 
+        let rtg = ref tg in 
+          ISet.iter (fun x -> 
+            greatest := !greatest + 1;
+            Hashtbl.add to_map x (!greatest);
+            begin match SrkUtil.Int.Map.find_opt x !assertions with (* if x in assertions, new vertex also in assertions *)
+              | Some assert_value -> 
+                assertions := SrkUtil.Int.Map.add !greatest assert_value !assertions 
+              | None -> () end;
+            rtg := WG.add_vertex !rtg !greatest
+            ) vertices;
+          ISet.iter (fun x -> 
+              rtg := WG.fold_succ_e (fun (u, w, v) tg' -> 
+                  WG.add_edge tg' (Hashtbl.find to_map u) w (Hashtbl.find to_map v)) 
+                !rtg x !rtg) vertices;
+          (!rtg, Hashtbl.find to_map)
+      in let remove_subgraph tg src = 
+        let vertices = dfs (-1, -1) tg src (fun _ _ _ -> ()) (ISet.add src ISet.empty) in 
+        ISet.fold (fun vtx tg' -> 
+          begin match SrkUtil.Int.Map.find_opt vtx !assertions with 
+          (* remove vertex from assertions map, new copied vertices already in it *)
+          | Some _ -> 
+            assertions := SrkUtil.Int.Map.remove vtx !assertions 
+          | None -> ()
+          end;
+          WG.remove_vertex tg' vtx) vertices tg in 
+      let inline_one tg (src, dst) (call_x, call_y) (call_src, call_dst) = 
+        let (tg, to_map) = copy_subgraph tg src in 
+        let tg = WG.add_edge tg call_x (Weight T.one) (to_map src) in 
+        let tg = WG.add_edge tg (to_map dst) (Weight T.one) call_y in 
+        let tg = WG.remove_edge tg call_x call_y in
+        let callees = PHT.find pmap (call_src, call_dst) in
+        let callers = PHT.find qmap (src, dst) in 
+        (* remove (src, dst) from callees list in caller *) 
+        PHT.add pmap (call_src, call_dst) (PS.remove (src, dst) callees);
+        (* remove (call_src, call_dst) from callers list *)
+        PHT.add qmap (src, dst) (PPS.remove ((call_src, call_dst),(call_x, call_y)) callers);
+        tg
+      in let rec do_inline depth tg =
+        if depth == 0 then tg else 
+          let sinks = compute_sinks () in 
+          let aux currproc tg = (* inline currproc into procedures that call it, assuming currproc is inlined. *)
+            let inline_targets = PHT.find qmap currproc in 
+              let tg' = 
+                PPS.fold (fun (call_proc, (call_x, call_y)) tg -> 
+                  inline_one tg currproc (call_x, call_y) call_proc
+                ) inline_targets tg in 
+                let (src, _) = currproc in 
+                  remove_subgraph tg' src  
+          in
+            displayer tg;
```

Comment:
Displayer is helpful for debugging the inliner, but I'd remove it from the interface now.